src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,263 @@
+var registerComponent = require('../core/component').registerComponent;
+var bind = require('../utils/bind');
+var isControllerPresent = require('../utils/tracked-controls').isControllerPresent;
+
+var TOUCH_CONTROLLER_MODEL_BASE_URL = 'https://cdn.aframe.io/controllers/oculus/oculus-touch-controller-';
+var TOUCH_CONTROLLER_MODEL_OBJ_URL_L = TOUCH_CONTROLLER_MODEL_BASE_URL + 'left.obj';
+var TOUCH_CONTROLLER_MODEL_OBJ_MTL_L = TOUCH_CONTROLLER_MODEL_BASE_URL + 'left.mtl';
+var TOUCH_CONTROLLER_MODEL_OBJ_URL_R = TOUCH_CONTROLLER_MODEL_BASE_URL + 'right.obj';
+var TOUCH_CONTROLLER_MODEL_OBJ_MTL_R = TOUCH_CONTROLLER_MODEL_BASE_URL + 'right.mtl';
+
+var GAMEPAD_ID_PREFIX = 'Oculus Touch';
+
+var FAKE_TOUCH_THRESHOLD = 0.00001;
+
+/**
+ * Oculus Touch Controls Component
+ * Interfaces with Oculus Touch controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and highlights the pressed buttons
+ */
+module.exports.Component = registerComponent('oculus-touch-controls', {
+  schema: {
+    hand: {default: 'left'},
+    buttonColor: { default: '#FAFAFA' },  // Off-white.
+    buttonHighlightColor: {default: '#22D1EE'},  // Light blue.
+    model: {default: true},
+    rotationOffset: {default: 0} // no default offset; -999 is sentinel value to auto-determine based on hand
+  },
+
+  // buttonId
+  // 0 - thumbstick
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
+    'left': {
+      axis0: 'thumbstick',
+      axis1: 'thumbstick',
+      button0: 'thumbstick',
+      button1: 'trigger',
+      button2: 'grip',
+      button3: ['A-or-X', 'X'],
```

Comment:
that is used so `hand-controls` and other higher level functions don't have to respond to both A and X separately just because they are named differently for each hand.